### ISSUE TITLE
Support configurable scan paths (--include / per-linter includePatterns)

### ISSUE THREAD

**Issue by aRustyDev** [carlrannaberg/cclint#3]
## Summary

`includePatterns` in `cclint.config.js` adds paths to **all** linters globally. This means if you add both `context/agents` and `context/commands` to `includePatterns`, the agents linter scans command files (and vice versa), producing false errors.

Projects that store context files outside `.claude/` (e.g. `context/agents/`, `context/commands/`) need a way to tell each linter where to look independently.

## Current Behavior

```js
// cclint.config.js
export default {
  rules: {
    includePatterns: ['context/agents', 'context/commands'],
  },
};
```

Result: the agents linter scans `context/commands/*.md` and flags them for missing `name` (required for agents, not commands).

### Option A: Per-linter include patterns

```js
export default {
  agentSchema: {
    includePaths: ['context/agents'],
  },
  commandSchema: {
    includePaths: ['context/commands'],
  },
};
```

### Option B: CLI --include flag

```bash
cclint agents --include context/agents
cclint commands --include context/commands
```

## Use Case

This repo is a Claude Code context marketplace with the structure:
```
context/
  agents/     ← 130+ agent definitions
  commands/   ← 23 command definitions
.claude/
  agents/     ← project-local agents
  commands/   ← project-local commands
```

Both directories contain valid Claude Code files and need linting, but they must be routed to the correct linter.

## Current Workaround

Symlinking `context/agents` into `.claude/agents/_context-agents` and running with `--follow-symlinks`. Works but is fragile and requires setup.
